wai-extra/Network/Wai/Test.hs

```diff
@@ -110,6 +111,9 @@ initState = ClientState Map.empty
 runSession :: Session a -> Application -> IO a
 runSession session app = ST.evalStateT (runReaderT session app) initState
 
+runSessionWith :: ClientState -> Session a -> Application -> IO (a, ClientState)
```

Comment:
Can you add a Haddock comment here explaining why you'd use this vs `runSession`?